Vector_graphics_on_surfaces/include/CGAL/Vector_graphics_on_surfaces/locally_shortest_path.h

```diff
@@ -0,0 +1,4814 @@
+ *               If omitted, then centroid of `polygon` will be used.
+ * \return the polar coordinates of the points as a pair (distance, angle)
+ */
+template <class K, class PointRange_2>
+std::vector<std::pair<typename K::FT, typename K::FT>>
+convert_to_polar_coordinates(const PointRange_2& points,
+                             std::optional<typename K::Point_2> center = std::nullopt)
+{
+  std::vector<std::pair<typename K::FT, typename K::FT>> result;
+  // compute naively the center
+  if (center==std::nullopt)
+  {
+    Bbox_2 bb = bbox_2(points.begin(), points.end());
+    center = typename K::Point_2((bb.xmax()+bb.xmin())/2., (bb.ymax()+bb.ymin())/2);
+  }
+
+  bool is_closed = points.front()==points.back();
+  std::size_t nbp=points.size();
+  if (is_closed) --nbp;
+  for (std::size_t i=0; i<nbp; ++i)
+  {
+    const typename K::Point_2& pt = points[i];
+    typename K::FT d = approximate_sqrt(squared_distance(*center, pt));
+    typename K::FT polar = std::atan2((pt.y()-center->y())/* /d */, (pt.x()-center->x())/* /d */);
+    result.emplace_back(d, polar);
+    // std::cout << center->x()+d*std::cos(polar) << " " << center->x()+d*std::sin(polar) << " 0\n";
+  }
+  if (is_closed) result.push_back(result.front());
+
+  return result;
+}
+
+
+
+/*!
+ * \ingroup VGSFunctions
+ * computes the face location of each vertex of a 2D polygon on `tmesh`.
+ * The vertices of the polygon are given as a pair of direction and distance with respect to a point corresponding to `center` on `tmesh`.
+ * \tparam TriangleMesh a model of `FaceListGraph` and `EdgeListGraph`
+ * \tparam K a model of `Kernel` with `K::FT` being a floating point number type (float or double)
+ * \param center the location on `tmesh` used as reference. The y-axis used for coordinates is `halfedge(center.first, tmesh)`.
+ * \param directions contains the direction one need to move from `center` to reach each vertex of the polygon.
+ * \param lengths the distance one need to move from `center` along the direction at the same position in `directions` to reach each vertex of the polygon.
+ * \param tmesh input triangle mesh supporting the vertices of the output polygon
+ * \param solver container for the precomputed information. If not initialized, it will be initialized internally.
+ * \return a face location for each vertex of the polygon
+ * \todo add named parameters
+ * \todo polygon orientation is not handled in the function and should be done outside of the function for now
+ * \todo offer something better than a 2D vector for the direction
+ * \todo directly handle polyline?
+ * \todo why the first polygon vertex is duplicated by the function? (most probably for the example but it shouldn't be done here)
+ */
+template <class K, class TriangleMesh>
+std::vector<CGAL::Polygon_mesh_processing::Face_location<TriangleMesh,typename K::FT>>
+trace_geodesic_polygon(const CGAL::Polygon_mesh_processing::Face_location<TriangleMesh, typename K::FT> &center,
```

Comment:
I am wondering if `trace` is the appropriate verb.  I have the feeling that it just comes from the French verb `tracer`, but is that what we want here?